sycl/include/sycl/ext/oneapi/properties/property.hpp

```diff
@@ -0,0 +1,174 @@
+//==---------- properties.hpp --- SYCL extension property tooling ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// HOW-TO: Add new compile-time property
+//  1. Add a new enumerator to
+//     `sycl::ext::oneapi::experimental::detail::PropKind` representing the new
+//     property. Increment
+//     `sycl::ext::oneapi::experimental::detail::PropKind::PropKindSize`
+//  2. Define property key class with `value_t` that must be `property_value`
```

Comment:
can these new property keys be in other namespaces?

Reply:
Sure. There is no requirement on the namespace of the properties. It would probably be preferable to have them in a namespace under `sycl::ext::oneapi::experimental` while it is still experimental, but since it is user-facing it should not be a requirement from the implementation.